tr.R

```diff
@@ -0,0 +1,319 @@
+    floor(SUBJNO %% 9) == 0 & TREVAL == "INVESTIGATOR" &
+      VISITNUM == 10.1 ~ NA_real_,
+    floor(SUBJNO %% 8) == 0 &
+      (TREVALID == "RADIOLOGIST 1" | TREVALID == "RADIOLOGIST 2") &
+      VISITNUM == 10.1 ~ NA_real_,
+    VISITNUM == 3 ~ TRSTRESN + 5,
+    TRUE ~ TRSTRESN
+  ),
+  "TRORRES" = as.character(TRSTRESN),
+  "TRSTRESC" = TRORRES,
+  "TRORRESU" = "mm",
+  "TRSTRESU" = TRORRESU,
+  "TRGRPID" = "TARGET",
+  "TRTESTCD" = "DIAMETER",
+  "TRTEST" = "Diameter"
+)
+
+
+# SUMDIAM
+tr3sum <- tr3 %>%
+  group_by(
+    STUDYID, USUBJID, VISITNUM, VISIT, TREVAL,
+    TREVALID, TRDTC, TRDY, TRGRPID, TRORRESU,
+    TRSTRESU, SAFETY, EFFICACY
+  ) %>%
+  summarize(TRSTRESN = sum(TRSTRESN)) %>%
+  mutate(
+    "TRTESTCD" = "SUMDIAM",
+    "TRTEST" = "Sum of Diameter",
+    "TRORRES" = as.character(TRSTRESN),
+    "TRSTRESC" = TRORRES
+  )
+
+# Non-target Tumors
+
+ntar <- floor(runif(n = nrows, min = 11, max = 15))
+ntr3a <- bind_rows(
+  (tr2 %>% mutate("tar1" = ntar, "TRLNKID" = "NT01")),
+  (tr2 %>% mutate("tar1" = ntar + 1, "TRLNKID" = "NT02")),
+  (tr2 %>% mutate("tar1" = ntar + 2, "TRLNKID" = "NT03")),
+  (tr2 %>% mutate("tar1" = ntar + 3, "TRLNKID" = "NT04")),
+  (tr2 %>% mutate("tar1" = ntar + 4, "TRLNKID" = "NT05"))
+) %>%
+  mutate("TREVAL" = "INVESTIGATOR")
+
+ntar <- floor(runif(n = nrows, min = 11, max = 15))
+ntr3b <- bind_rows(
+  (tr2 %>% mutate("tar1" = ntar, "TRLNKID" = "R1-NT01")),
+  (tr2 %>% mutate("tar1" = ntar + 1, "TRLNKID" = "R1-NT02")),
+  (tr2 %>% mutate("tar1" = ntar + 2, "TRLNKID" = "R1-NT03")),
+  (tr2 %>% mutate("tar1" = ntar + 3, "TRLNKID" = "R1-NT04")),
+  (tr2 %>% mutate("tar1" = ntar + 4, "TRLNKID" = "R1-NT05"))
+) %>%
+  mutate(
+    "TREVAL" = "INDEPENDENT ASSESSOR",
+    "TREVALID" = "RADIOLOGIST 1"
+  )
+
+ntar <- floor(runif(n = nrows, min = 11, max = 15))
+ntr3c <- bind_rows(
+  (tr2 %>% mutate("tar1" = ntar, "TRLNKID" = "R2-NT01")),
+  (tr2 %>% mutate("tar1" = ntar + 1, "TRLNKID" = "R2-NT02")),
+  (tr2 %>% mutate("tar1" = ntar + 2, "TRLNKID" = "R2-NT03")),
+  (tr2 %>% mutate("tar1" = ntar + 3, "TRLNKID" = "R2-NT04")),
+  (tr2 %>% mutate("tar1" = ntar + 4, "TRLNKID" = "R2-NT05"))
+) %>%
+  mutate(
+    "TREVAL" = "INDEPENDENT ASSESSOR",
+    "TREVALID" = "RADIOLOGIST 2"
+  )
+
+ntr3 <- bind_rows(ntr3a, ntr3b, ntr3c)
+
+# Adding Non Target Result Values Randomly
+ntrorres <- c("NOT", "UNEQUIVOCAL", "ABSENT", "PRESENT")
+
+ntrespd <- as.data.frame(ntrorres)
+ntrespd$ntresn <- NULL
+ntrespd$ntresn <- row.names(ntrespd)
+
+ntrows <- dim(ntr3)[1]
+x <- runif(ntrows, 0, 1)
+ntr3$ntresn <- as.numeric(cut(x, breaks = c(0, 0.05, 0.15, 0.6, 1)))
+
+ntr3 <- merge(ntr3, ntrespd, by = "ntresn") %>%
+  mutate(
+    "TRGRPID" = "NON-TARGET",
+    "TRTESTCD" = "TUMSTATE",
+    "TRTEST" = "Tumor State",
+    "TRORRES" = case_when(
+      VISITNUM == 3 ~ "PRESENT",
+      !is.na(ntrorres) & ntrorres != "NOT" ~ ntrorres
+    ),
+    "SUBJNO" = as.numeric(substr(USUBJID, 8, 12))
+  )

```

Comment:
When sourcing the script this call fails with
```
Error: must be a character vector, not a `factor` object
```
I think it is due to `ntrorres` created as factor in line 178.

Could you check please?

Reply:
Thank you for reviewing.  Moved the files to the correct locations. I am not getting the error when running the tr.R script. But, I updated the logic and hope it will fix the error on your side. Please let me know if you are still getting the error.

Reply:
Unfortunately, I am still getting an error.

It works when I replace line 178 with
```
ntrespd <- tibble::tibble(ntrorresf)
```

Reply:
Thank you for suggesting. Updated tr.R with tibble code.  Also added tidyselect to all the three scripts (tu, tr, rs).  Please let me know if you notice any other issues.